components/neeva/renderer/content_filtering_agent.cc

```diff
@@ -133,6 +133,13 @@ ContentFilteringPolicy ContentFilteringAgent::GetPolicyForRequest(
     return ContentFilteringPolicy::kAllow;
   }
 
+  // There's no point in blocking foo.com from making requests from first-party origins (e.g. foo.com). 
+  // Doing so can break logins or other site functionality. 
+  if (!IsThirdParty(url, first_party_origin)) {
+    // Block cookies for first-party cookies to stop them from tracking users too. 
```

Comment:
nit: "Block cookies for first-party cookies" doesn't read quite right.

I think the point to make here is that we are downgrading `mode` to `BLOCK_COOKIES` in this case to minimize impact to sites.